server/index.js

```diff
@@ -37,6 +38,23 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const productId = parseInt(req.params.productId, 10);
+
+  const sql = 'SELECT * FROM "products" WHERE "productId" = $1';
+  const params = [productId];
+
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next(new ClientError(`cannot ${req.method} ${req.originalUrl}`, 404));
```

Comment:
A better message would be something like "cannot find product with id ${productId}"

Reply:
Right on, I'll make a clearer message